src/common/src/types/mod.rs

```diff
@@ -236,6 +273,126 @@ impl DataType {
         }
     }
 
+    pub fn type_len(&self) -> i16 {
+        match self {
+            DataType::Boolean => 1,
+            DataType::Int16 => 2,
+            DataType::Int32 | DataType::Float32 | DataType::Date => 4,
+            DataType::Int64
+            | DataType::Float64
+            | DataType::Timestamp
+            | DataType::Timestampz
+            | DataType::Time => 8,
+            DataType::Decimal
+            | DataType::Varchar
+            | DataType::Interval
+            | DataType::Struct(_)
+            | DataType::List { .. } => -1,
+        }
+    }
+
+    // NOTE:
+    // Refer https://github.com/postgres/postgres/blob/master/src/include/catalog/pg_type.dat when add new TypeOid.
+    // Be careful to distinguish oid from array_type_oid.
+    // Such as:
+    //  https://github.com/postgres/postgres/blob/master/src/include/catalog/pg_type.dat#L347
+    //  For Numeric(aka Decimal): oid = 1700, array_type_oid = 1231
+    pub fn from_oid(oid: i32) -> crate::error::Result<Self> {
```

Comment:
How about moving all these Postgres stuffs to a new file (submodule), and it's allowed to `impl DataType` as well? I guess that'll help us to maintain.